packages/dx-grid-core/src/plugins/virtual-table-state/helpers.ts

```diff
@@ -43,12 +43,17 @@ export const mergeRows: MergeRowsFn = (
 };
 
 export const calculateRequestedRange: CalculateRequestedRangeFn = (
-  virtualRows, newRange, pageSize,
+  virtualRows, newRange, pageSize, referenceIndex,
 ) => {
   const loadedInterval = intervalUtil.getRowsInterval(virtualRows);
   const isAdjacentPage = Math.abs(loadedInterval.start - newRange.start) < 2 * pageSize;
-  if (isAdjacentPage) {
-    return intervalUtil.difference(newRange, loadedInterval);
+  const calculatedRange = intervalUtil.difference(newRange, loadedInterval);
+  if (isAdjacentPage && calculatedRange !== intervalUtil.empty) {
+    if (calculatedRange.start - referenceIndex > pageSize / 2) {
+      calculatedRange.start -= pageSize;
```

Comment:
In non-infinite scrolling mode request range shouldn't be affected

Reply:
You're right, I'll fix it.